src/wallet/wallet2.cpp

```diff
@@ -2626,7 +2644,19 @@ void wallet2::process_new_transaction(const crypto::hash &txid, const cryptonote
           {
             size_t idx = i->second;
             THROW_WALLET_EXCEPTION_IF(idx >= m_transfers.size(), error::wallet_internal_error, "Output tracker cache index out of range");
-            m_transfers[idx].m_uses.push_back(std::make_pair(height, txid));
+
+            if (m_track_uses)
+              m_transfers[idx].m_uses.push_back(std::make_pair(height, txid));
+
+            // We'll re-process all txs which *might* be spends when we disable
+            // background sync and retrieve the spend key. We don't know if an
+            // output is a spend in this tx if we don't know its key image.
+            if (m_background_sync_enabled && !m_transfers[idx].m_key_image_known && m_background_sync_cache.find(txid) == m_background_sync_cache.end())
+            {
+              size_t bgs_cache_idx = m_background_sync_cache.size();
+              LOG_PRINT_L2("Adding plausible spent tx " << txid << " to background sync cache (idx=" << bgs_cache_idx << ")");
+              m_background_sync_cache.insert({txid, background_syncd_tx{bgs_cache_idx, tx, o_indices, height, ts, double_spend_seen}});
+            }
```

Comment:
Info that may help fellow reviewers and testers:

This case is not easy to hit because in most situations a transaction has either an enote going into the wallet, or the wallet knows already one of the enotes that serve as inputs for a transaction, both cases which are then handled elsewhere.

The easiest way that I found to trigger this particular case while testing "manually":

* Create a new wallet and make copies of the wallet files
* Let the first still empty wallet go into background sync
* Send a tx to the wallet address to fund, wait until unlocked
* Spend the funds using the copied wallet files, using `sweep_single` instead of transfer to avoid change
* The first wallet in background sync will go through this code as soon as the spend tx gets mined

Reply:
If you're testing with the RPC wallet, the following should trigger it as well:

- Create a new wallet
- Receive to the wallet
- Sweep all funds out of the wallet (or use `sweep_single`)
- Enable background sync mode via the new `/set_background_sync` RPC
- Hard rescan via `/rescan_blockchain`
    - The wallet should consider the sweep tx a "plausible spent tx" when scanning

The functional tests do this.